lib/chef/provider/group/gpasswd.rb

```diff
@@ -22,6 +22,7 @@ class Chef
   class Provider
     class Group
       class Gpasswd < Chef::Provider::Group::Groupadd
+        provides :group, platform: %w(suse default)
```

Comment:
There's no `default` platform.   This should probably be `os: linux` which via your herirarchy of specificity would get overridden by more precise `platform` or `platform_version` overrides.  Although this might simply be `provides :group` without any specificity since many OSen would be getting this via the default in the platform_map (everything that doesn't declare a group provider needs this provider).